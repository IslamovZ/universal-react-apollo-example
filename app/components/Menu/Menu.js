import React from 'react';
import { DropdownButton, MenuItem } from 'react-bootstrap';
import { Link } from 'react-scroll';
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';
import { connect } from 'react-redux';

// import MenuItemLink from '../MenuItemLink/MenuItemLink';
import s from './menu';


const Menu = ({ data }) => {
  const { loading } = data;

  if (loading) {
    return (null);
  }

  const { wp_query } = data;
  const { menu } = wp_query;

  return (
    <div>
      <ul className={s.menu}>
        {menu.map((menuItem) => {
          return (
            <li key={menuItem.id}><Link to={menuItem.title.toLowerCase()} smooth>{menuItem.title}</Link></li>
          );
        })}
      </ul>
      <DropdownButton id="dropdown" bsStyle="default" noCaret title={[<span />]} className={s.dropdown}>
        {menu.map((menuItem) => {
          return (
            <MenuItem key={menuItem.id} className={s['dropdown-menu']}>
              <Link to={menuItem.title.toLowerCase()} smooth>{menuItem.title}</Link>
            </MenuItem>
          );
        })}
      </DropdownButton>
    </div>
    );
};

const MenuWithData = graphql(gql`
  query getMenu ($lang: String!) {
    wp_query {
      menu (name: $lang) {
        id
        title
      }
    }
  }
`, {
  options: ({ language }) => ({
    variables: { lang: language || 'gb' },
  }),
}
)(Menu);

const MenuWithDataAndState = connect(
  state => ({ language: state.language })
)(MenuWithData);

export default MenuWithDataAndState;

