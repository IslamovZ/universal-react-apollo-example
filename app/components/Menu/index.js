import React from 'react';

import { NavMenu, DropdownMenu } from 'components';

const Menu = () => (
  <div>
    <NavMenu />
    <DropdownMenu />
  </div>
);

export default Menu;

