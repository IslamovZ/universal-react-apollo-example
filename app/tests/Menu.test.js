import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/lib/test-utils';

import Menu from 'components/Menu/Menu';

describe('Menu Component', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <MockedProvider><Menu /></MockedProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
