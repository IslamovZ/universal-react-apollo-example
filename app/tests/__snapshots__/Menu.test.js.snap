exports[`Menu Component renders correctly 1`] = `
<div>
  <div>
    Loading....
  </div>
  <div
    className="dropdown btn-group btn-group-default">
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-default"
      disabled={false}
      id="dropdown"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      type="button">
      <span />
    </button>
    <ul
      aria-labelledby="dropdown"
      className="dropdown-menu"
      role="menu" />
  </div>
</div>
`;
